### PR TITLE
Remove redshift async test job

### PR DESCRIPTION
**Pull request by pankajastro**
Address https://github.com/apache/airflow/pull/28850#issuecomment-1467778448

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -399,6 +399,8 @@ def write_version(filename: str = str(AIRFLOW_SOURCES_ROOT / "airflow" / "git_ve
     "wheel",
     "yamllint",
     "aioresponses",
+    # TODO: why??
+    "aiobotocore>=2.1.1",
```
Comment (eladkal) [inline setup.py:406]: I suggest to sync with AWS before doing further work.
We are not yet sure if we should add this lib
https://github.com/apache/airflow/pull/30032#discussion_r1132975173

Comment (eladkal) [reply setup.py:406]: cc @shubham22 @kaxil

Comment (pankajastro) [reply setup.py:406]: Hi @eladkal, I agree with your concern. I was testing the workaround suggested by @potiuk here https://github.com/apache/airflow/pull/28850#issuecomment-1468033661 to run the test

**Review by potiuk — APPROVED** [review]
I treat @o-nikolas approval on #29923 as general OK from the AWS team to follow this approach.

File: setup.py (hunk 1 above)
Comment (shubham22) [reply setup.py:406]: I see that this is already merged, but can we please wait until 3/22 before making more deferrable contributions? Recent changes from @potiuk in https://github.com/apache/airflow/pull/30144 do alleviate many concerns, but there is always a chance that users won't be able to access new Boto functionality. We are in the process of sharing the raised concerns internally at AWS. However, I do understand that there is no better alternative than using aiobotocore and tradeoff is not bad after the CI job separation. cc: @syedahsn @o-nikolas

Comment (potiuk) [reply setup.py:406]: One more thing here. Technically with the job implemented as it is right now wiht my #30144 we have another possibility. We **could** reverse the approach. 

Currently we remove `aiobotocore` in the separate job and upgrade to latest boto and run all the relevant provider tests. 

But we could do it the opposite:  we could use latest boto/botocore in "main", remove aibotocore from the "devel" set of dependencies and install aiobotocore (together with downgrading boto/botocore) in the separate job.

This has slightly different characteristics:

* by default the constraints we publish will not have aiobotocore and willhave botocore/boto not compatible with the latest aiobotocore - this means that if someone would like to use deferrable operators from AWS would have to deliberately install aiobotocore and downgrade boto/botocore with it.
* the separate job would test automatically if the "aiobotocore" compatible boto still works with all the unit-tested provider operators - this way any PR that would be based on newer features would fail at PR time.
* then we could make a deliberate decision that this is ok (and add conditional skips in the unit tests) if that happens

So basically this is this trade-off:
1. (current) deferrable operators work out-of-the-box with the "official constraints" (but without latest boto/botocore)
2. (possible) - deferrable operators will not work with official constraints and they require deliberately installing aiobotocore (and downgrading boto/botocore) - but latest botocore is used in those constraints

Those are the two trade-offs, and we can still change the decision (easily) which way go. With #30144  this is as easy as changing few lines in the CI scripts.

Comment (potiuk) [reply setup.py:406]: Also I just created o#30161 that adds flexibility in switching between the two choices by separating aiobotocore to be separate extra for Airflow. This also makes sure that (in the current approach) aiobotocore is installed in the PROD image of airflow - which makes deferrable operators work truly "out-of-the-box" in the released Airflow image (I forgot about it while commenting on #30127). 

This also shows another consequence that you should be aware @o-nikolas @shubham22 @syedahsn

* If we go for 1) then deferrable operators will work out-of-the-box with the regular PROD image of airflow (with older botocore/boto). Users will be able to build their image with the latest botocore with something like that:

```
FROM apache/airflow:2.6.0-python3.9
RUN pip uninstall aiobotocore && pip install --upgrade botocore boto # no constraints here, they will be incompatible
```

But they loose deferrable operators this way.


* If we go for 2) then deferrable operators will not work out-of-the-box but the botocore/boto will be latest at the moment of release. Users will have to do something like that (this will downgrade botocore) and build their image (or equivalent way of building the image for your service)

```
FROM apache/airflow:2.6.0-python3.9
RUN pip install aiobotocore==2.5.0 # no constraints here, they will be incompatible
```

I hope this will keep you better informed on the choices we have - happy to hear which of the two is your preference. I am ok with both approaches (and there is basically no other way to handle it).

Comment (o-nikolas) [reply setup.py:406]: Personally I prefer the approach of shipping with aiobotocore/async support by default. I think this has two benefits:
1. A more complete functioning product ships by default, the user doesn't need to do anything specific to use deferrable operators. It is more user friendly.
1. I prefer the scenario where boto/botocore version is restricted from the outset. This sets a better expectation of what features will be available for all usecases. The opposite approach allows users build dependencies against current boto features and then if they enable async they lose those features. This is a bad and confusing user experience (or at least, worse of the two options IMHO).

I'm curious to hear what others think.

Comment (syedahsn) [reply setup.py:406]: I think shipping aiobotocore by default is the better option. It will be a smoother user experience if users don't have to change their workflows to use deferrables if their workflows use features not covered by aiobotocore. It's not an ideal situation in either case, but this option is less likely to cause issues for users wanting to use deferrable operators.

Comment (kaxil) [reply setup.py:406]: I am with @o-nikolas on this one -- a case where everything works out of the box is better than if a user has to make choices depending.

Also, we want to increase and drive the adoption of Deferrable operators, and decreasing road-blocks for user's adopting it would be my preference

Comment (shubham22) [reply setup.py:406]: Agree with the thoughts expressed by @o-nikolas @syedahsn and @kaxil. Priority should be given to providing a better out-of-the-box user experience. If we don't make aiobotocore the default option, it will increase the workload for users, and as Niko mentioned, it may require refactoring of DAGs once the user enables async. On the other hand, we don't have any evidence yet that restricting the botocore version will cause significant customer problems. 

Thank you @potiuk  for spending time on this and providing us with options. We're certainly converging to a better approach than what we started with.

Comment (potiuk) [reply setup.py:406]: Yeah. I did not want to bias it, but that was also my personal preference (but I wanted to objectively show you the options). It seems much easier to explain as well, and I have a feeling that deferring new features by few months is not a huge issue as the adoption is usually anyhow delayed and deferrable operators not being available by default would be a big bummer. 

With the #30161 now merged, the aiobotocore will also be included in the PROD image by default, so we are on track to increase the adoption of deferrable operators :).

Comment (pankajastro) [reply setup.py:406]: This is awesome, look like we have solution now. Thank you so much @potiuk for stepping-up bringing it in better shape it was long since pending 🚀
